code/datums/components/harvester.dm

```diff
@@ -156,6 +166,27 @@
 		target.adjustStaminaLoss(-30)
 		target.heal_overall_damage(6, 0, updating_health = TRUE)
 
+///Updates the color of the weapon based on what chem is loaded in. Updates both the item sprite and mob sprite
+/datum/component/harvester/proc/update_loaded_color(datum/source)
+	SIGNAL_HANDLER
+	var/obj/item/item_parent = parent
+	var/image/item_overlay = image('icons/obj/items/vali.dmi', item_parent, "[initial(item_parent.icon_state)]_loaded")
+	if(!loaded_reagent)
+		item_overlay.color = COLOR_GREEN
+	else
+		item_overlay.color = initial(loaded_reagent.color)
+	item_parent.overlays.Add(item_overlay)
```

Comment:
you're misusing this signal/applying overlays incorrectly.

Look at where this signal is coming from, you need to add the overlay to the list its sending as an arg, not directly add it.